source/common/signal/fatal_error_handler.cc

```diff
@@ -22,6 +24,29 @@ ABSL_CONST_INIT static absl::Mutex failure_mutex(absl::kConstInit);
 using FailureFunctionList = std::list<const FatalErrorHandlerInterface*>;
 ABSL_CONST_INIT std::atomic<FailureFunctionList*> fatal_error_handlers{nullptr};
 
+// Use an atomic operation since on fatal error we'll consume the
+// fatal_action_manager and don't want to have any locks as they aren't
+// async-signal-safe.
+ABSL_CONST_INIT std::atomic<FatalAction::FatalActionManager*> fatal_action_manager{nullptr};
+ABSL_CONST_INIT std::atomic<Envoy::Server::Instance*> server_instance{nullptr};
```

Comment:
It looks like these are only ever set together. Would it make sense to put them in a struct referenced by a single pointer?

Reply:
🤦 yes, I originally meant for the FatalActionMananger to own the server ref.